code/game/objects/items/documents.dm

```diff
@@ -115,3 +221,218 @@
+	if(print_time == 1 SECONDS)
+		print_time = 5 SECONDS
+		message += "SLOW."
+	else
+		print_time = 1 SECONDS
+		message += "LIGHTNING FAST."
+	balloon_alert(user, message)
+
+/obj/item/inspector/clown/proc/cycle_sound(mob/user)
+	print_sound_mode++
+	if(print_sound_mode > max_mode)
+		print_sound_mode = INSPECTOR_PRINT_SOUND_MODE_NORMAL
+	balloon_alert(user, "You turn the dial with holes in it, setting the device's bleep setting to [mode_names[print_sound_mode]] mode.")
+
+/obj/item/inspector/clown/create_slip()
+	var/obj/item/paper/fake_report/slip = new(get_turf(src))
+	slip.generate_report(get_area(src))
+
+/**
+ * # Bananium HONK-spect scanner
+ *
+ * An upgraded version of the fake N-spect scanner
+ *
+ * Can print things way faster, at full power the reports printed by this will destroy
+ * themselves and leave water behind when folding is attempted by someone who isn't an
+ * origami master. Printing at full power costs INSPECTOR_POWER_USAGE_HONK cell units
+ * instead of INSPECTOR_POWER_USAGE_NORMAL cell units.
+ */
+/obj/item/inspector/clown/bananium
+	name = "\improper Bananium HONK-spect scanner"
+	desc = "Honkmother-blessed inspection device. Performs inspections according to Clown protocols when activated, then \
+			prints a clowncrypted report regarding the maintenance of the station. Hard to replace."
+	icon_state = "bananium_inspector"
+	w_class = WEIGHT_CLASS_SMALL
+	max_mode = BANANIUM_CLOWN_INSPECTOR_PRINT_SOUND_MODE_LAST
+	///How many more times can we print?
+	var/paper_charges = 200
+	///Max value of paper_charges
+	var/max_paper_charges = 500
+	///How much charges are restored per paper consumed
+	var/charges_per_paper = 25
+
+/obj/item/inspector/clown/bananium/proc/check_settings_legality()
+	if((print_sound_mode == INSPECTOR_PRINT_SOUND_MODE_NORMAL)&&(print_time < 1 SECONDS))
+		if(cell.use(power_to_speak))
+			say("Setting combination forbidden by Geneva convention revision CCXXIII selected, reverting to defaults")
+		print_time = 5 SECONDS
+		print_sound_mode = INSPECTOR_PRINT_SOUND_MODE_CLASSIC
+		power_per_print = INSPECTOR_POWER_USAGE_NORMAL
+
+/obj/item/inspector/clown/bananium/attackby(obj/item/I, mob/user, params)
+	..()
+	check_settings_legality()
+	if(istype(I, /obj/item/paper/fake_report)||(paper_charges>=max_paper_charges))
+		to_chat(user, "<span class='info'>\The [src] refuses to consume \the [I]!</span>")
+	else if(istype(I, /obj/item/paper))
+		to_chat(user, "<span class='info'>\The [src] consumes \the [I]!</span>")
+		paper_charges+=charges_per_paper
+		if(paper_charges>max_paper_charges)
+			paper_charges = max_paper_charges
+		qdel(I)
+
+/obj/item/inspector/clown/bananium/Initialize()
+	. = ..()
+	playsound(src, 'sound/effects/angryboat.ogg', 150, FALSE)
+
+/obj/item/inspector/clown/bananium/create_slip()
+	if(print_time == 0.1 SECONDS)
+		var/obj/item/paper/fake_report/water/slip = new(get_turf(src))
+		slip.generate_report(get_area(src))
+	else
+		..()
```

Comment:
In line of the previous suggestions, I'd add this one:

```suggestion
/obj/item/inspector/clown/bananium/create_slip()
	if(print_time == 0.1 SECONDS)
		var/obj/item/paper/fake_report/water/slip = new(get_turf(src))
		slip.generate_report(get_area(src))
		return
	return ..()
```

However, `if(print_time == 0.1 SECONDS)` is not a good check. What you should actually be checking here is whether it's on the clown mode, which should aggregate both the low print time and the water slips.
Can you change this so it's more clear?